charts/replicated-library/templates/lib/_init_container.tpl

```diff
@@ -0,0 +1,69 @@
+{{- /* The main container included in the main */ -}}
+{{- define "replicated-library.initContainer" -}}
+  {{- $values := "" -}}
+  {{- if and (hasKey . "ContextValues") (hasKey .ContextValues "app") -}}
+    {{- $values = .ContextValues.app -}}
+  {{- else -}}
+    {{- fail "_init_container.tpl requires the 'app' ContextValues to be set" -}}
+  {{- end -}}
+  {{- $_ := set $.ContextValues "names" (dict "context" "app") -}}
+{{- range $containerName, $containerValues := $values.initContainers -}}
+- name: {{ default "container" $containerName }}
```

Comment:
I don't think that would be allowed. However, I think its implicit here that this default statement would never default to `container` because for the range to work you have to have the key name present in the first place i think. Like you couldn't set:

```yaml
app:
  example:
    containers:
      "":
        image: ...
```

So we could probably just change this `default` statement and hardcode to `$containerName`?

Reply:
Yea sounds good if the default should never trigger let's just use $containerName. I'd rather it error if somehow we get no name than the more confusing error of including 2 contaienrs which would template but not install.